djangocms_frontend/contrib/modal/cms_plugins.py

```diff
@@ -0,0 +1,119 @@
+from cms.plugin_pool import plugin_pool
+from django.utils.translation import gettext_lazy as _
+
+from ... import settings
+from ...cms_plugins import CMSUIPlugin
+from ...common.attributes import AttributesMixin
+from .. import modal
+from . import forms, models
+
+mixin_factory = settings.get_renderer(modal)
+
+
+@plugin_pool.register_plugin
+class ModalPlugin(mixin_factory("Modal"), AttributesMixin, CMSUIPlugin):
+    """
+    Component > "Modal" Plugin
+    https://getbootstrap.com/docs/5.0/components/modal/
+    """
+
+    name = _("Modal")
+    module = _("Frontend")
+    model = models.Modal
+    form = forms.ModalForm
+    change_form_template = "djangocms_frontend/admin/modal.html"
+    allow_children = True
+    child_classes = [
+        "ModalTriggerPlugin",
+        "ModalContainerPlugin",
+    ]
+
+    fieldsets = [
+        (None, {"fields": ("modal_siblings",)}),
+    ]
+
+
+@plugin_pool.register_plugin
+class ModalTriggerPlugin(mixin_factory("ModalTrigger"), AttributesMixin, CMSUIPlugin):
+    """
+    Component > "Modal" Plugin
+    https://getbootstrap.com/docs/5.0/components/modal/
+    """
+
+    name = _("Modal trigger")
+    module = _("Frontend")
+    model = models.ModalTrigger
+    form = forms.ModalTriggerForm
+    allow_children = True
+    parent_classes = [
+        "CardPlugin"
```

Comment:
I am not sure why a modal trigger should be part of a card? I am having difficulties imagining a static button or other trigger that should trigger a modal. Can you give a use-case example? Things that come to my mind are typically related to a POST request, e.g., form submissions (Say, for confirmations)

Reply:
You're right, @fsbraun. The parent class should be "ModalPlugin", not CardPlugin. 
The trigger for the collapse could be everywhere on the cms page. I thought that the ModalTrigger should be inside the ModalPlugin to force an organised page tree. Or do you think it is better to remove this dependency?